### PR TITLE
Feat GATv2 model

### PR DESCRIPTION
**Pull request by AliGhadirii**

### CODE REVIEW

File: models/GATv2.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        heads,
+        num_classes,
+        dropout_rate=None,
+        last_sigmoid=False,
+    ):
+
+        super(GATv2, self).__init__()
+        self.num_layers = len(dim_shapes)
+        self.last_sigmoid = last_sigmoid
+        self.dropout_rate = dropout_rate
```
Comment (afroozsheikh) [inline models/GATv2.py:22]: Are you sure that it is logical to use dropouts in GAT?

File: models/GATv2.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        self.convs = nn.ModuleList()
+        for l in range(self.num_layers):
+            if l == 0:
+                self.convs.append(
+                    GATv2Conv(dim_shapes[l][0], dim_shapes[l][1], heads=heads)
+                )
+            else:
+                self.convs.append(
+                    GATv2Conv(dim_shapes[l][0] * heads, dim_shapes[l][1], heads=heads)
```
Comment (afroozsheikh) [inline models/GATv2.py:40]: i have no idea about that multiplication to heads. Are you sure about it?

File: models/GATv2.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                )
+
+        self.pooling = global_mean_pool
+
+        self.classifier = nn.Sequential(
+            nn.Linear(heads * dim_shapes[-1][1], 16),
```
Comment (afroozsheikh) [inline models/GATv2.py:46]: من یه جا دیدم همینجا دراپ اوت رو گذاشته بود. یعنی:
        self.post_mp = nn.Sequential(
            nn.Linear(hidden_dim, hidden_dim), nn.Dropout(0.25), 
            nn.Linear(hidden_dim, output_dim))

File: models/GATv2.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            x = F.dropout(x, p=self.dropout_rate)
+        x = self.classifier(x)
+
+        if self.last_sigmoid:
+            return torch.sigmoid(x)
+
+        return x
+
+    def load_weights(self, path):
+        self.load_state_dict(torch.load(path))
```
Comment (afroozsheikh) [inline models/GATv2.py:76]: عالی بودد :]
